silkworm/core/execution/processor.cpp

```diff
@@ -284,10 +280,18 @@ CallResult ExecutionProcessor::call(const Transaction& txn, const std::vector<st
         state_.set_nonce(*sender, state_.get_nonce(*txn.sender()) + 1);
     }
 
-    if (!evm().bailout) {
-        const intx::uint256 required_funds = protocol::compute_call_cost(txn, effective_gas_price, evm_);
-        state_.subtract_from_balance(*txn.sender(), required_funds);
+    intx::uint256 required_funds{0};
+    if (evm().bailout) {
+        required_funds = protocol::compute_call_required_funds(txn, effective_gas_price, evm_);
+        // If the bailout option is on, add the required funds to the sender's balance
+        // so that after the transaction costs are deducted, the sender's balance is unchanged.
+        state_.add_to_balance(*txn.sender(), required_funds);
+    } else {
+        required_funds = protocol::compute_call_cost(txn, effective_gas_price, evm_);
     }
+
+    SILKWORM_ASSERT(protocol::validate_call_funds(txn, evm_, state_.get_balance(*txn.sender())) == ValidationResult::kOk);
+    state_.subtract_from_balance(*txn.sender(), required_funds);
```

Comment:
I still don't understand why we are subtracting different values depending on the bailout on/off. This still feels quite a hack.

Reply:
Yep, I hear you about the feeling. I guess the reason is `standard_transfer` function [here](https://github.com/erigontech/silkworm/blob/ci/bailout_refactor/silkworm/core/execution/evm.hpp#L83) does apply the bailout on each value transfer included the `txn.value` from `sender`. @lupin012 can you comment on this?

Reply:
Now, Yes in standard_transfer() called in EVM::create() EVM::Call() in case of NOT bailout the value is subtracked. Each txn (also internal) must subtracks its ether transfer

Reply:
> I still don't understand why we are subtracting different values depending on the bailout on/off. This still feels quite a hack.

Also this point has been superseded by the last refactoring.